tests/jax/components/building/datasets_test.py

```diff
@@ -0,0 +1,192 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for TransitionDataset and TrajectoryDataset classes for Jax-based Mava"""
+
+from dataclasses import dataclass
+from types import SimpleNamespace
+from typing import Any, Callable, Dict, Optional
+
+import pytest
+import reverb
+from tensorflow.python.framework import dtypes, ops
+
+from mava import specs
+from mava.adders import reverb as reverb_adders
+from mava.components.jax.building.datasets import TrajectoryDataset, TransitionDataset
+from mava.systems.jax.builder import Builder
+from tests.jax.mocks import make_fake_env_specs
+
+env_spec = make_fake_env_specs()
+Transform = Callable[[reverb.ReplaySample], reverb.ReplaySample]
+
+
+def adder_signature_fn(
+    ma_environment_spec: specs.MAEnvironmentSpec,
+    extras_specs: Dict[str, Any],
+) -> Any:
+    """Signature function that helps in building a simple server"""
+    return reverb_adders.ParallelNStepTransitionAdder.signature(
+        ma_environment_spec=ma_environment_spec, extras_specs=extras_specs
+    )
+
+
+class MockBuilder(Builder):
+    def __init__(self) -> None:
+        """Creates a mock builder for testing"""
+        self.simple_server = reverb.Server(
+            [
+                reverb.Table.queue(
+                    name="table_0",
+                    max_size=100,
+                    signature=adder_signature_fn(env_spec, {}),
+                )
+            ]
+        )
+        data_server_client = SimpleNamespace(
+            server_address=f"localhost:{self.simple_server.port}"
+        )
+        trainer_id = "table_0"
+        self.store = SimpleNamespace(
+            data_server_client=data_server_client, trainer_id=trainer_id
+        )
+
+
+@pytest.fixture
+def mock_builder() -> MockBuilder:
+    """Create builder mock"""
+    return MockBuilder()
+
+
+@dataclass
+class TransitionDatasetConfigTest:
+    sample_batch_size: int = 512
+    prefetch_size: Optional[int] = None
+    num_parallel_calls: int = 24
+    max_in_flight_samples_per_worker: Optional[int] = None
+    postprocess: Optional[Transform] = None
+
+
+@dataclass
+class TrajectoryDatasetConfigTest:
+    sample_batch_size: int = 512
+    max_in_flight_samples_per_worker: int = 1024
+    num_workers_per_iterator: int = -2
+    max_samples_per_stream: int = -2
+    rate_limiter_timeout_ms: int = -2
+    get_signature_timeout_secs: Optional[int] = None
+
+
+def test_init_transition_dataset() -> None:
+    """Test initiator of TransitionDataset component"""
+    transition_dataset = TransitionDataset(config=TransitionDatasetConfigTest)  # type: ignore
```

Comment:
Please change this to a fixture (as we did in other tests such as the reverb_compoents_test) then it can be of type `TrajectoryDatasetConfig` and you can remove the type: ignore and make it multiline so that it is not > 88 characters as per the pre-commit checks

Reply:
The error from mypy here is not about the length of the line but it is for `Argument "config" to "TransitionDataset" has incompatible type "TransitionDatasetConfigTest"; expected "TransitionDatasetConfig"` which means that the dataconfig we gave should be of type TransitionDatasetConfig.
I will try to make mock and update the values from TransitionDatasetConfig inside it!